zenpy/lib/api.py

```diff
@@ -1067,6 +1068,22 @@ def comments(self, ticket, include_inline_images=False):
         """
         return self._query_zendesk(self.endpoint.comments, 'comment', id=ticket, include_inline_images=repr(include_inline_images).lower())
 
+    @extract_id(Ticket, TicketComment)
+    def comment_redact(self, ticket, comment, text) -> TicketComment:
+        """
+        Redact text from ticket comment.
+
+        :param ticket: Ticket object or id
+        :param comment: Comment object or id
+        :param text: Text to be redacted from comment
+        :return Comment: Ticket Comment object
+        """
+
+        url = self._build_url(self.endpoint.comments.redact(ticket, comment))
+        redacted_comment = self._put(url, {'text': text})
+        redacted_comment._set_dirty()
```

Comment:
Just curious, why do you need to mark the comment as dirty?

Reply:
We do not need it. I was inspired by the show_macro_effect method before I was sure what was return on the endpoint, as the API does not describe the successful return, only the 400 bas request return.